geode-core/src/main/java/org/apache/geode/cache/query/internal/index/PartitionedIndex.java

```diff
@@ -175,13 +176,16 @@ public List getBucketIndexes() {
    */
   public Index getBucketIndex() {
     Index index = null;
-    synchronized (this.bucketIndexes) {
+    try {
       if (this.bucketIndexes.size() > 0) {
         List<Index> indexList = this.bucketIndexes.values().iterator().next();
         if (indexList != null && indexList.size() > 0) {
           index = indexList.get(0);
         }
       }
+    } catch (ConcurrentModificationException | NullPointerException e) {
```

Comment:
Any reason bucketIndexes can't just be a ConcurrentHashMap rather than a SynchronizedMap?

Reply:
We probably could change that as well.

In this case we still need to protect the returned value too (a list as the value)